File: src/main/java/ar/edu/unq/epers/unidad5/model/PrecioPromedio.java
package ar.edu.unq.epers.unidad5.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@ToString
@Setter
@Getter
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)

@Document
public class PrecioPromedio {
    @Id
    private String codigo;
    private double value = 0;
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/dto/PrecioPromedio.kt
package ar.edu.unq.epers.unidad5.dto

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
class PrecioPromedio() {
    @Id
    var codigo: String? = null
    var value:Double = 0.toDouble()
}